ccl/changefeedccl: TestAlterChangefeedDatabaseQualifiedNames failed

ccl/changefeedccl.TestAlterChangefeedDatabaseQualifiedNames [failed](https://mesolite.cluster.engflow.com/invocations/default/bb2ec9f1-bfc5-480c-9265-5eaaf6c578f9?testReportRun=1&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [d36cdb971710d69d43c9f242d45019ddea71049b](https://github.com/cockroachdb/cockroach/commits/d36cdb971710d69d43c9f242d45019ddea71049b):


```
=== RUN   TestAlterChangefeedDatabaseQualifiedNames
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestAlterChangefeedDatabaseQualifiedNames2792751203
    test_log_scope.go:82: use -show-logs to present logs inline
    alter_changefeed_test.go:1120: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterChangefeedDatabaseQualifiedNames2792751203
--- FAIL: TestAlterChangefeedDatabaseQualifiedNames (6.44s)
=== RUN   TestAlterChangefeedDatabaseQualifiedNames/cloudstorage
    helpers_test.go:1541: making server as secondary tenant
    helpers_test.go:1620: making cloudstorage feed factory
    helpers_test.go:1293: forcing DB level changefeed for *changefeedccl.cloudFeedFactory - CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES drivers WITH OPTIONS (resolved = '100ms', diff, initial_scan = 'yes')
    alter_changefeed_test.go:1104: error executing query="ALTER CHANGEFEED 1130544214415966209 ADD d.users WITH initial_scan UNSET diff" args=[]: pq: cannot alter targets for a database level changefeed
--- FAIL: TestAlterChangefeedDatabaseQualifiedNames/cloudstorage (6.43s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedDatabaseQualifiedNames.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57569

ccl/changefeedccl.TestAlterChangefeedDatabaseQualifiedNames [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20829962?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20829962?buildTab=artifacts#/) on master @ [cd5829b3a7d1a2390532a54ac90ad9bcab9b95c5](https://github.com/cockroachdb/cockroach/commits/cd5829b3a7d1a2390532a54ac90ad9bcab9b95c5):


```
=== RUN   TestAlterChangefeedDatabaseQualifiedNames
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/f236cfce7e32f4b492de4b4f1cd80c96/logTestAlterChangefeedDatabaseQualifiedNames1995601214
    test_log_scope.go:82: use -show-logs to present logs inline
    alter_changefeed_test.go:1120: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f236cfce7e32f4b492de4b4f1cd80c96/logTestAlterChangefeedDatabaseQualifiedNames1995601214
--- FAIL: TestAlterChangefeedDatabaseQualifiedNames (7.16s)
=== RUN   TestAlterChangefeedDatabaseQualifiedNames/cloudstorage
    helpers_test.go:1541: making server as secondary tenant
    helpers_test.go:1620: making cloudstorage feed factory
    helpers_test.go:1348: did not force enriched envelope for CREATE CHANGEFEED FOR d.drivers WITH resolved = '100ms', diff because *changefeedccl.cloudFeedFactory is not supported
    helpers_test.go:1293: forcing DB level changefeed for *changefeedccl.cloudFeedFactory - CREATE CHANGEFEED FOR DATABASE d INCLUDE TABLES drivers WITH OPTIONS (resolved = '100ms', diff, initial_scan = 'yes')
    alter_changefeed_test.go:1104: error executing query="ALTER CHANGEFEED 1131046969864355841 ADD d.users WITH initial_scan UNSET diff" args=[]: pq: cannot alter targets for a database level changefeed
--- FAIL: TestAlterChangefeedDatabaseQualifiedNames/cloudstorage (7.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedDatabaseQualifiedNames.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/alter_changefeed_test.go
+++ b/pkg/ccl/changefeedccl/alter_changefeed_test.go
@@ -680,7 +680,9 @@ func TestAlterChangefeedErrors(t *testing.T) {
 		sqlDB := sqlutils.MakeSQLRunner(s.DB)
 		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY)`)
 		sqlDB.Exec(t, `CREATE TABLE bar (a INT PRIMARY KEY)`)
-		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo`)
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo`, optOutOfMetamorphicDBLevelChangefeed{
+			reason: "db level changefeeds don't support ADD/DROP TARGETS in ALTER CHANGEFEEDs",
+		})
 		defer closeFeed(t, testFeed)
 
 		feed, ok := testFeed.(cdctest.EnterpriseTestFeed)
@@ -1087,7 +1089,10 @@ func TestAlterChangefeedDatabaseQualifiedNames(t *testing.T) {
 		sqlDB.Exec(t, `CREATE TABLE d.users (id INT PRIMARY KEY, name STRING)`)
 		sqlDB.Exec(t, `INSERT INTO d.drivers VALUES (1, 'Alice')`)
 		sqlDB.Exec(t, `INSERT INTO d.users VALUES (1, 'Bob')`)
-		testFeed := feed(t, f, `CREATE CHANGEFEED FOR d.drivers WITH resolved = '100ms', diff`)
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR d.drivers WITH resolved = '100ms', diff`,
+			optOutOfMetamorphicDBLevelChangefeed{
+				reason: "db level changefeeds don't support ADD/DROP TARGETS in ALTER CHANGEFEEDs",
+			})
 		defer closeFeed(t, testFeed)
 
 		assertPayloads(t, testFeed, []string{
@@ -2381,7 +2386,9 @@ func TestAlterChangefeedAddTargetsDuringBackfill(t *testing.T) {
 
 		registry := s.Server.JobRegistry().(*jobs.Registry)
 		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo
-WITH resolved = '100ms', min_checkpoint_frequency='1ns'`)
+WITH resolved = '100ms', min_checkpoint_frequency='1ns'`, optOutOfMetamorphicDBLevelChangefeed{
+			reason: "db level changefeeds don't support ADD/DROP TARGETS in ALTER CHANGEFEEDs",
+		})
 
 		g := ctxgroup.WithContext(context.Background())
 		g.Go(func() error {

--- a/pkg/ccl/changefeedccl/protected_timestamps_test.go
+++ b/pkg/ccl/changefeedccl/protected_timestamps_test.go
@@ -381,7 +381,10 @@ func TestChangefeedAlterPTS(t *testing.T) {
 		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
 		sqlDB.Exec(t, `CREATE TABLE foo2 (a INT PRIMARY KEY, b STRING)`)
 		f2 := feed(t, f, `CREATE CHANGEFEED FOR table foo with protect_data_from_gc_on_pause,
-			resolved='1s', min_checkpoint_frequency='1s'`)
+			resolved='1s', min_checkpoint_frequency='1s'`,
+			optOutOfMetamorphicDBLevelChangefeed{
+				reason: "db level changefeeds don't support ADD/DROP TARGETS in ALTER CHANGEFEEDs",
+			})
 		defer closeFeed(t, f2)
 
 		getNumPTSRecords := func() int {

